clusterloader2/pkg/test/simpletestexecutor.go

```diff
@@ -28,20 +34,80 @@ func createSimpleTestExecutor() TestExecutor {
 
 // ExecuteTest executes test based on provided configuration.
 func (ste *simpleTestExecutor) ExecuteTest(ctx Context, conf *api.Config) error {
+	defer cleanupResources(ctx)
+	err := ctx.GetFramework().CreateAutomanagedNamespaces(int(conf.AutomanagedNamespaces))
+	if err != nil {
+		return fmt.Errorf("automanaged namespaces creation failed: %v", err)
+	}
+
+	for i := range conf.Steps {
+		if err := ste.ExecuteStep(ctx, &conf.Steps[i]); err != nil {
+			return err
+		}
+	}
 	return nil
 }
 
 // ExecuteStep executes single test step based on provided step configuration.
 func (ste *simpleTestExecutor) ExecuteStep(ctx Context, step *api.Step) error {
-	return nil
+	var err error
+	if len(step.Measurements) > 0 {
+		// TODO: handle measurments
+	} else {
+		var wg wait.Group
+		var lock sync.Mutex
+		for i := range step.Phases {
+			phase := &step.Phases[i]
+			wg.Start(func() {
+				if phaseErr := ste.ExecutePhase(ctx, phase); phaseErr != nil {
+					lock.Lock()
+					defer lock.Unlock()
+					if err == nil {
+						err = phaseErr
+					}
+				}
+			})
+		}
+		wg.Wait()
+	}
+	return err
 }
 
 // ExecutePhase executes single test phase based on provided phase configuration.
 func (ste *simpleTestExecutor) ExecutePhase(ctx Context, phase *api.Phase) error {
+	if phase.NamespaceRange == nil {
+		for j := range phase.ObjectBundle {
+			ste.ExecuteObject(ctx, "", phase.ReplicasPerNamespace, &phase.ObjectBundle[j])
```

Comment:
TBH - I'm not sure what semantics ReplicasPerNamespace implies for non-namespaced objects. This is not a blocker for us (as I don't think we create any non-namespaced objects in our tests) - but sth we need to think about.

cc @wojtek-t